mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/api/persistence/value/ValueResultJsonSerializer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.persistence.value;
+
+import org.mule.runtime.api.value.ImmutableValueResult;
+import org.mule.runtime.api.value.Value;
+import org.mule.runtime.api.value.ValueResult;
+import org.mule.runtime.extension.api.values.ImmutableValue;
+import org.mule.runtime.extension.internal.persistence.DefaultImplementationTypeAdapterFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Serializer that can convert a {@link ValueResult} type into a readable and processable
+ * JSON representation and from a JSON {@link String} to an {@link ValueResult} instance
+ *
+ * @since 1.0
+ */
+public class ValueResultJsonSerializer {
+
+  private final Gson gson;
+  private final DefaultImplementationTypeAdapterFactory valueResultTypeAdapterFactory =
+      new DefaultImplementationTypeAdapterFactory<>(ValueResult.class, ImmutableValueResult.class);
+  private final DefaultImplementationTypeAdapterFactory valueTypeAdapterFactory =
+      new DefaultImplementationTypeAdapterFactory<>(Value.class, ImmutableValue.class);
+
+  public ValueResultJsonSerializer() {
```

Comment:
> private and static create method
At least in all the serializers that I'm seeing, the convention is to use constructors and not factory methods.